test/test_retrieve_filepath.py

```diff
@@ -0,0 +1,69 @@
+"""
```

Comment:
@MrCreosote This is ready for a re-review. I finally added tests here that I think cover all the cases. I also extracted the relevant function into its own module.